idmclient/client.go

```diff
@@ -79,36 +79,46 @@ func (c *basicAuthClient) DoWithBody(req *http.Request, r io.ReadSeeker) (*http.
 // httpbakey.Client.VisitWebPage to perform an OAuth login interaction.
 func UbuntuSSOOAuthVisitWebPage(client *http.Client, tok *usso.SSOData) func(u *url.URL) error {
 	return func(u *url.URL) error {
-		return ussoOAuthVisit(client, tok, u)
+		lm, err := LoginMethods(client, u)
+		if err != nil {
+			return err
+		}
+		if lm.UbuntuSSOOAuth == "" {
+			return errgo.New("Ubuntu SSO OAuth login not supported")
+		}
+		return UbuntuSSOOAuthVisit(client, lm.UbuntuSSOOAuth, tok, u)
 	}
 }
 
-func ussoOAuthVisit(client *http.Client, tok *usso.SSOData, u *url.URL) error {
+// LoginMethods returns the available login methods for the given url.
```

Comment:
This should document what kind of URLs are appropriate to use with this function, I think.